patches/minecraft/net/minecraft/block/TNTBlock.java.patch

```diff
@@ -0,0 +1,126 @@
+          if (p_220082_2_.func_175640_z(p_220082_3_)) {
+-            func_196534_a(p_220082_2_, p_220082_3_);
++            createExplosion(p_220082_2_, p_220082_3_);
+             p_220082_2_.func_217377_a(p_220082_3_, false);
+          }
+ 
+@@ -40,7 +55,7 @@
+ 
+    public void func_220069_a(BlockState p_220069_1_, World p_220069_2_, BlockPos p_220069_3_, Block p_220069_4_, BlockPos p_220069_5_, boolean p_220069_6_) {
+       if (p_220069_2_.func_175640_z(p_220069_3_)) {
+-         func_196534_a(p_220069_2_, p_220069_3_);
++         createExplosion(p_220069_2_, p_220069_3_);
+          p_220069_2_.func_217377_a(p_220069_3_, false);
+       }
+ 
+@@ -48,7 +63,7 @@
+ 
+    public void func_176208_a(World p_176208_1_, BlockPos p_176208_2_, BlockState p_176208_3_, PlayerEntity p_176208_4_) {
+       if (!p_176208_1_.func_201670_d() && !p_176208_4_.func_184812_l_() && p_176208_3_.func_177229_b(field_212569_a)) {
+-         func_196534_a(p_176208_1_, p_176208_2_);
++         createExplosion(p_176208_1_, p_176208_2_);
+       }
+ 
+       super.func_176208_a(p_176208_1_, p_176208_2_, p_176208_3_, p_176208_4_);
+@@ -56,16 +71,24 @@
+ 
+    public void func_180652_a(World p_180652_1_, BlockPos p_180652_2_, Explosion p_180652_3_) {
+       if (!p_180652_1_.field_72995_K) {
+-         TNTEntity tntentity = new TNTEntity(p_180652_1_, (double)((float)p_180652_2_.func_177958_n() + 0.5F), (double)p_180652_2_.func_177956_o(), (double)((float)p_180652_2_.func_177952_p() + 0.5F), p_180652_3_.func_94613_c());
++         TNTEntity tntentity = tntEntityCreator.create(p_180652_1_, (double)((float)p_180652_2_.func_177958_n() + 0.5F), (double)p_180652_2_.func_177956_o(), (double)((float)p_180652_2_.func_177952_p() + 0.5F), p_180652_3_.func_94613_c());
+          tntentity.func_184534_a((short)(p_180652_1_.field_73012_v.nextInt(tntentity.func_184536_l() / 4) + tntentity.func_184536_l() / 8));
+          p_180652_1_.func_217376_c(tntentity);
+       }
+    }
+ 
++   /**
++    * @deprecated Prefer using the local method
++    */
++   @Deprecated
+    public static void func_196534_a(World p_196534_0_, BlockPos p_196534_1_) {
+       func_196535_a(p_196534_0_, p_196534_1_, (LivingEntity)null);
+    }
+ 
++   /**
++    * @deprecated Prefer using the one that mods can override
++    */
++   @Deprecated
+    private static void func_196535_a(World p_196535_0_, BlockPos p_196535_1_, @Nullable LivingEntity p_196535_2_) {
+       if (!p_196535_0_.field_72995_K) {
+          TNTEntity tntentity = new TNTEntity(p_196535_0_, (double)((float)p_196535_1_.func_177958_n() + 0.5F), (double)p_196535_1_.func_177956_o(), (double)((float)p_196535_1_.func_177952_p() + 0.5F), p_196535_2_);
+@@ -74,13 +97,25 @@
+       }
+    }
+ 
++   public void createExplosion(World p_196534_0_, BlockPos p_196534_1_) {
++      createExplosion(p_196534_0_, p_196534_1_, null);
++   }
++
++   private void createExplosion(World p_196535_0_, BlockPos p_196535_1_, @Nullable LivingEntity p_196535_2_) {
```

Comment:
Make this protected/public to make it so mods can override the one that takes an entity as well.